zipline/assets/_securities.pyx

```diff
@@ -97,14 +97,14 @@ cdef class Security:
         elif isinstance(x, int):
             x_as_int = x
         else:
-            return NotImplemented
+            return False
```

Comment:
Do you have reason to believe that "False" rather than "NotImplemented" is correct? The documentation is bad and sparse, but what I could find suggested that NotImplemented was the correct thing to return when the __richcmp__ function doesn't know how to do the comparison. And it passes the test I wrote to confirm that it doesn't crash.